src/redux/stores/collection/collectionApi.ts

```diff
@@ -1,54 +1,46 @@
+import { getTokenFromContract } from '@airswap/metadata';
+import { TokenInfo } from '@airswap/typescript';
 import { Web3Provider } from '@ethersproject/providers';
-import ERC721 from '@openzeppelin/contracts/build/contracts/ERC721.json';
 import { createAsyncThunk } from '@reduxjs/toolkit';
-import { Contract, ethers } from 'ethers';
 
 import { CollectionToken } from '../../../entities/CollectionToken/CollectionToken';
-import { transformErc721TokenToCollectionToken } from '../../../entities/CollectionToken/CollectionTokenTransformers';
-import { Erc721Token } from '../../../entities/Erc721Token/Erc721Token';
-
-export const getCollectionErc721Contract = (library: Web3Provider, token: string): Contract => {
-  const erc721Interface = new ethers.utils.Interface(ERC721.abi);
-
-  return new Contract(token, erc721Interface, library);
-};
+import { transformNFTTokenToCollectionToken } from '../../../entities/CollectionToken/CollectionTokenTransformers';
 
 interface fetchNFTMetadataParams {
   library: Web3Provider,
   collectionToken: string,
   startIndex: number,
 }
 
-export const fetchNFTMetadata = createAsyncThunk<
-CollectionToken[], fetchNFTMetadataParams>(
+export const fetchCollectionTokens = createAsyncThunk<(
+CollectionToken | undefined)[], fetchNFTMetadataParams>(
   'collection/fetchNFTMetadata',
   async ({ library, collectionToken, startIndex }) => {
-    // TODO: Add support for ERC-1155
-    const collectionContract = getCollectionErc721Contract(library, collectionToken);
-
-    if (!collectionContract) {
-      throw new Error('No collection contract found');
-    }
-
     const CHUNK_SIZE = 20;
     const tokensToFetch = new Array(CHUNK_SIZE)
       .fill(null)
       .map((value, index) => startIndex + index);
 
     const dataPromises = tokensToFetch.map(async (tokenId) => {
-      const tokenURI = await collectionContract.tokenURI(tokenId);
-      const res = await fetch(
-        tokenURI.replace('ipfs://', 'https://ipfs.io/ipfs/'),
-      );
-      const token = await res.json() as Erc721Token;
-
-      return transformErc721TokenToCollectionToken(
-        token,
-        tokenId,
-        0.154,
-      );
+      let tokenInfo: TokenInfo;
+
+      try {
+        tokenInfo = await getTokenFromContract(library, collectionToken, tokenId.toString());
+      } catch (e) {
+        console.error(new Error(`Unable to fetch data for ${collectionToken} with id ${tokenId}`));
```

Comment:
This can't happen as that promise can't/doesn't reject

Reply:
This is intentional, we don't want the whole createAsyncThunk to reject if one token is not found. Here's a scenario:

A collection has a total of 30 tokens, id's go from 1 until 30. Right now `CHUNK_SIZE` is 20, so when tokens 20 - 40 gets fetched with `fetchCollectionTokens` it will fail for token id 30 until 40. But we still want the 20 until 30 to go to the store.

That's why the return type of `fetchCollectionTokens` is `(`CollectionToken | undefined)[]`.

This might change when we get to the irregular token id ticket. https://github.com/airswap/airswap-marketplace/issues/49 Then we could define a limit of the token id's.